desktop/wallet/main.js

```diff
@@ -22,6 +26,20 @@ function handleLatestUpdate(newVersion) {
   win.webContents.send('new-version', newVersion);
 }
 
+async function getPublicKey() {
+  await TransportNodeHid.create();
+
+  // todo: move this to a shared fn
+  let accountIndexHex = accountIndex.toString(16);
+  if (accountIndexHex.length === 1) {
+    accountIndexHex = '0' + accountIndexHex;
+  }
+  console.log('accountIndexHex', accountIndexHex); // TODO
+  const publicKey = await transport.exchange(Buffer.from('800400' + accountIndexHex + '00', 'hex'));
+  console.log('publicKey', publicKey); // TODO
+  return win.webContents.send('public-key-received', publicKey);
+}
```

Comment:
by moving the node stuff into `main.js`, we are able to avoid issues with `fs` being undefined. We can communicate via ipc messageBus, as shown.